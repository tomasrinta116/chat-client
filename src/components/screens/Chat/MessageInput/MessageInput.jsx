import React from "react";
import { withStyles } from "@material-ui/core/styles";
import TextField from "@material-ui/core/TextField";
import InputAdornment from "@material-ui/core/InputAdornment";
import SendIcon from "@material-ui/icons/SendRounded";
import IconButton from "@material-ui/core/IconButton";
import styles from "./MessageInput.styles";

class MessageInput extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      message: ""
    };
  }

  handleChange = event => {
    this.setState({ message: event.target.value });
  };

  sendMessage = (channel, message) => {
    channel.sendUserMessage(message, (message, error) => {
      if (error) return console.log(error);
    });
  };

  handleClick = () => {
    const { channel, addMessage } = this.props;
    const { message } = this.state;
    this.sendMessage(channel, message);
    let newMessage = { _sender: { userId: "You" }, message: message };
    addMessage(newMessage);
    this.setState({ message: "" });
  };

  render() {
    const {
      classes: { messageInputContainer, messageInputTextField }
    } = this.props;
    const { message } = this.state;
    let color;

    !message ? (color = "default") : (color = "primary");

    return (
      <div className={messageInputContainer}>
        <TextField
          multiline
          rowsMax="4"
          value={message}
          onChange={this.handleChange}
          className={messageInputTextField}
          margin="normal"
          variant="outlined"
          InputProps={{
            endAdornment: (
              <InputAdornment position="end">
                <IconButton
                  variant="contained"
                  onClick={this.handleClick}
                  color={color}
                >
                  <SendIcon />
                </IconButton>
              </InputAdornment>
            )
          }}
        />
      </div>
    );
  }
}

export default withStyles(styles)(MessageInput);
