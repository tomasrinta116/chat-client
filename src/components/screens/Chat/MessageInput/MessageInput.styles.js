const styles = {
  messageInputContainer: {
    display: "flex",
    alignItems: "center",
    justifyContent: "center",
    margin: "auto"
  },
  messageInputTextField: {
    width: "60%"
  }
};

export default styles;
