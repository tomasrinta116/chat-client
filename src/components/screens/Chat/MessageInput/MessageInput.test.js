import React from "react";
import { shallow } from "enzyme";
import MessageInput from "./MessageInput";
import TextField from "@material-ui/core/TextField";

describe("<MessageInput />", () => {
  test("it renders correctly", () => {
    const wrapper = shallow(<MessageInput />).dive();
    console.log(wrapper.debug());
    expect(wrapper).toMatchSnapshot();
  });

  test("inputting a message updates component state", () => {
    const wrapper = shallow(<MessageInput />).dive();
    const textField = wrapper.find(TextField);
    expect(textField).toHaveLength(1);
    textField.simulate("change", { target: { value: "test message" } });
    expect(wrapper.state().message).toEqual("test message");
  });
});
