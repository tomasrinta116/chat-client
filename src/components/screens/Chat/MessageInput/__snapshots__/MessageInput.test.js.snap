// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MessageInput /> it renders correctly 1`] = `
<div
  className="MessageInput-messageInputContainer-1"
>
  <TextField
    InputProps={
      Object {
        "endAdornment": <WithStyles(InputAdornment)
          position="end"
        >
          <WithStyles(IconButton)
            color="default"
            onClick={[Function]}
            variant="contained"
          >
            <pure(SendRoundedIcon) />
          </WithStyles(IconButton)>
        </WithStyles(InputAdornment)>,
      }
    }
    className="MessageInput-messageInputTextField-2"
    margin="normal"
    multiline={true}
    onChange={[Function]}
    required={false}
    rowsMax="4"
    select={false}
    value=""
    variant="outlined"
  />
</div>
`;
